src/Type/Php/SubstrDynamicReturnTypeExtension.php

```diff
@@ -40,13 +46,43 @@ public function getTypeFromFunctionCall(
 
 			$negativeOffset = IntegerRangeType::fromInterval(null, -1)->isSuperTypeOf($offset)->yes();
 			$zeroOffset = (new ConstantIntegerType(0))->isSuperTypeOf($offset)->yes();
+			$length = null;
 			$positiveLength = false;
 
 			if (count($args) === 3) {
 				$length = $scope->getType($args[2]->value);
 				$positiveLength = IntegerRangeType::fromInterval(1, null)->isSuperTypeOf($length)->yes();
 			}
 
+			$constantStrings = TypeUtils::getConstantStrings($string);
+			if (count($constantStrings) > 0 &&
+				$offset instanceof ConstantIntegerType &&
+				($length === null || $length instanceof ConstantIntegerType)) {
+				$results = [];
+				foreach ($constantStrings as $constantString) {
+					if ($length !== null) {
+						$substr = substr(
+							$constantString->getValue(),
+							$offset->getValue(),
+							$length->getValue(),
+						);
+					} else {
+						$substr = substr(
+							$constantString->getValue(),
+							$offset->getValue(),
+						);
+					}
+
+					if (is_bool($substr)) {
```

Comment:
should this php8+ only error be put into a baseline? if so in which one? :)

Reply:
Most likely build/baseline-8.0.neon

Reply:
thanks, works like a charm.